cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/GetRevisionTaskThreaded.java

```diff
@@ -121,6 +122,8 @@ public GetRevisionTaskThreaded(CouchDB sourceDb,
                 DocumentRevsList>(executorService, this.requests, threads + 1) {
             @Override
             public DocumentRevsList executeRequest(BulkGetRequest request) {
+                // since this is part of a thread pool, we'll rename each thread as it takes a task.
+                Thread.currentThread().setName("GetRevisionThread: "+GetRevisionTaskThreaded.this.sourceDb.getIdentifier());
```

Comment:
We should maybe catch the `SecurityException` that could be thrown by the call to `setName()` and not set a custom thread name in that case. I don't think `setName()` could fail on Android, but it probably could on desktop Java with a security manager.